File: InterviewPreparationKit/RecursiveDigitSum.py
#!/bin/python3
# https://www.hackerrank.com/challenges/recursive-digit-sum/problem
import math
import os
import random
import re
import sys

# Complete the superDigit function below.
def superDigit(n, k):
    if len(n) == 1 and k == 1:
        return int(n)

    sum = 0
    for i in range(len(n)):
        sum += int(n[i]) * k

    return superDigit(str(sum), 1)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    nk = input().split()

    n = nk[0]

    k = int(nk[1])

    result = superDigit(n, k)

    fptr.write(str(result) + '\n')

    fptr.close()

File: InterviewPreparationKit/RecursiveDigitSum.java
// https://www.hackerrank.com/challenges/recursive-digit-sum/problem
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the superDigit function below.
    static int superDigit(String n, int k) {
        if (n.length() == 1 && k == 1) {
            return Integer.parseInt(n);
        }
        
        long sum = 0L;
        for (int i = 0; i < n.length(); i++) {
            sum += ((long) n.charAt(i) - 48) * (long) k;
        }

        return superDigit(sum + "", 1);
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String[] nk = scanner.nextLine().split(" ");

        String n = nk[0];

        int k = Integer.parseInt(nk[1]);

        int result = superDigit(n, k);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}
